src/components/App.tsx

```diff
@@ -1374,6 +1376,19 @@ class App extends React.Component<ExcalidrawProps, AppState> {
         isCollaborating: true,
         isLoading: opts.showLoadingState ? true : this.state.isLoading,
       });
+
+      try {
+        const elements = await loadFromFirebase(roomId, roomSecret);
+        if (elements) {
+          updateScene(
+            { type: "SCENE_INIT", payload: { elements } },
+            { init: true },
```

Comment:
This always overrides the history and doesn't do the merging, hence creating the bug above. Why did you put that here?

Reply:
removed `init: true`